### DISCUSSION TITLE
List files and directories created/updated by `quarto render`

### DISCUSSION THREAD

**Discussion by Victor-Botelho** [quarto-dev/quarto-cli#2372] [Feature Requests]
My team currently uses a Git Hook to automatically render `.qmd` files and stage and commit the files and directories created/updated by such command. It would be interesting if `quarto render` could return some kind of output containing all files and directories updated/created, so I could program my Git Hook to staged and commit such results.

**Comment by mavam** [comment]
I was literally about to ask the same question. My use case is just slightly different: I would like to conditionally run Quarto only when the inputs have changed, in a Makefile in my project:

```make
quarto-r-deps:
	@Rscript -e 'if (!require(pak)) { \
                 install.packages("pak"); \
                 library(pak) \
               }; \
               pak::local_install_dev_deps()'

quarto: quarto-r-deps
	@quarto render
```

If I had some sort of manifest file that contains a list of generated files, then I can use that as a target with all `*.qmd` files as inputs and create the make dependency DAG easily.

**Reply by Victor-Botelho** [reply]
The log behavior is even stranger when you look at the logs generated by the `quarto render` command. Even if you ask for the maximum level of verbosity when converting multiple files, Quartos' log will indicate only one output created: that of the first file rendered. It completely ignores any `_freeze/` and `*_files/` directories, as well as other files created besides the first one rendered.

**Comment by cderv** [comment]
Just as reference, possible related use case in https://github.com/quarto-dev/quarto-r/issues/14
